apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/ResponseField.kt

```diff
@@ -140,236 +87,16 @@ open class ResponseField internal constructor(
-        conditions: List<Condition>?
-    ): ResponseField {
-      return ResponseField(
-          type = Type.OBJECT,
-          responseName = responseName,
-          fieldName = fieldName,
-          arguments = arguments.orEmpty(),
-          optional = optional,
-          conditions = conditions.orEmpty()
-      )
-    }
-
-    /**
-     * Factory method for creating a Field instance representing [Type.LIST].
-     *
-     * @param responseName alias for the result of a field
-     * @param fieldName name of the field in the GraphQL operation
-     * @param arguments arguments to be passed along with the field
-     * @param optional whether the arguments passed along are optional or required
-     * @param conditions list of conditions for this field
-     * @return Field instance representing [Type.LIST]
-     */
-    @JvmStatic
-    fun forList(
-        responseName: String,
-        fieldName: String,
-        arguments: Map<String, Any?>?,
-        optional: Boolean,
-        conditions: List<Condition>?
-    ): ResponseField {
-      return ResponseField(
-          type = Type.LIST,
-          responseName = responseName,
-          fieldName = fieldName,
-          arguments = arguments.orEmpty(),
-          optional = optional,
-          conditions = conditions.orEmpty()
-      )
-    }
-
-    /**
-     * Factory method for creating a Field instance representing a custom GraphQL Scalar type, [Type.CUSTOM]
-     *
-     * @param responseName alias for the result of a field
-     * @param fieldName name of the field in the GraphQL operation
-     * @param arguments arguments to be passed along with the field
-     * @param optional whether the arguments passed along are optional or required
-     * @param customScalar the custom scalar type of the field
-     * @param conditions list of conditions for this field
-     * @return Field instance representing [Type.CUSTOM]
-     */
-    @JvmStatic
-    fun forCustomScalar(
-        responseName: String,
-        fieldName: String,
-        arguments: Map<String, Any?>?,
-        optional: Boolean,
-        customScalar: CustomScalar,
-        conditions: List<Condition>?
-    ): CustomScalarField {
-      return CustomScalarField(
-          responseName = responseName,
-          fieldName = fieldName,
-          arguments = arguments.orEmpty(),
-          optional = optional,
-          customScalar = customScalar,
-          conditions = conditions.orEmpty()
-      )
-    }
-
     @JvmStatic
     fun isArgumentValueVariableType(objectMap: Map<String, Any?>): Boolean {
       return (objectMap.containsKey(VARIABLE_IDENTIFIER_KEY)
           && objectMap[VARIABLE_IDENTIFIER_KEY] == VARIABLE_IDENTIFIER_VALUE && objectMap.containsKey(VARIABLE_NAME_KEY))
     }
+
+    fun Type.customScalarName(): String = when(this) {
+      is Type.NotNull -> ofType.customScalarName()
+      is Type.Named -> name
+      else -> error("Type '$this' is not a scalar type, check codegen")
+    }
```

Comment:
I'm a little confused what this function does. It returns just a raw type `name` right? Should we call it smth like `leafName` / `rawName`

Reply:
👍